src/interfaces/BarButton.ts

```diff
@@ -23,17 +34,6 @@ export interface BarButtonImage {
 }
 
 /**
- * A button in a navigation bar.
- */
-export type BarButton =
-  | BarButtonText
-  | BarButtonImage & {
-      /**
-       * Button ID
-       */
-      id: string;
-      /**
-       * A Boolean value that enables and disables the bar button.
-       */
-      disabled?: boolean;
-    };
+* A button in a navigation bar.
+*/
+export type BarButton = BarButtonImage | BarButtonText;
```

Comment:
tidied this up a little, I believe that all of these interfaces would be better in a single file